### DISCUSSION TITLE
Mutiny emitOn with request scoped beans

### DISCUSSION THREAD

**Discussion by madocx** [quarkusio/quarkus#37492] [Q&A] [answered]
So the Mutiny documentation has a very non-specific warning about the use of the emitOn method with non thread-safe objects such as request scoped cdi beans. However, I can't find any additional information about this warning. I'm hoping someone in the community can enlighten me as to what I should be careful with explicity.

> Be careful as this operator can lead to concurrency problems with non thread-safe objects such as CDI request-scoped beans.

Does the emit on effect upstream request scoped beans? Or is the concern downstream? Or is there no way to tell?

[a comment by an automated account is not shown]

**Comment by jponge** [comment]
It's a concern downstream, because `emitOn` shifts to a worker pool thread that might not carry context propagation, or you might have concurrent access on a CDI bean that you didn't anticipate. Your mileage varies, it's a just warning.

**Reply by madocx** [reply]
Interesting, so not something we should avoid using entirely, just be aware of the possibility of issues?

I have some cases where I'm using request scoped jaxrs resources and need to use emitOn to restore context on some chained events. I guess I could move the request scoped injections to method parameters if necessary and make the jaxrs resource application scoped.

**Reply by jponge** [reply, marked as answer]
Not something you have to avoid, just be mindful of the threading model underneath (see https://quarkus.io/guides/quarkus-reactive-architecture)

**Reply by madocx** [reply]
Excellent, thank you sir!
